spring-cloud-task-core/src/test/java/org/springframework/cloud/task/micrometer/TaskMetricsTests.java

```diff
@@ -43,15 +45,15 @@
 
 	@BeforeEach
 	public void before() {
-		Metrics.globalRegistry.getMeters().forEach(Metrics.globalRegistry::remove);
-		simpleMeterRegistry = new SimpleMeterRegistry();
-		Metrics.addRegistry(simpleMeterRegistry);
-		taskMetrics = new TaskMetrics();
+		this.simpleMeterRegistry = (SimpleMeterRegistry) new SimpleMeterRegistry().withTimerObservationHandler();
+		this.taskMetrics = new TaskMetrics(this.simpleMeterRegistry);
 	}
 
 	@AfterEach
 	public void after() {
-		Metrics.globalRegistry.getMeters().forEach(Metrics.globalRegistry::remove);
+		this.simpleMeterRegistry.clear();
+		this.simpleMeterRegistry.setCurrentObservation(null);
```

Comment:
Actually shouldn't the instrumentation handle setting the current observation to `null` ? I mean if you've done both `onTaskStartup` and `onTaskEnd` then there MUST not be any current observation. So instead of setting this to null you should do an assertion that current observation is null.